classes/travel/Patrol.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+class Patrol {
+    public int $id;
+    public int $start_time;
+    public ?int $travel_time;
+    public ?int $travel_interval;
+    public int $region_id;
+    public int $current_x;
+    public int $current_y;
+    public int $map_id = 1;
+    public string $name = "Patrol";
+    public int $level = 100;
+    public int $village_id;
+    public ?int $ai_id = null;
+    public string $patrol_type;
+    public string $alignment;
+    const DESTINATION_BUFFER_MS = 5000; // duration non-looped patrols should appear at their destination
+    public function __construct(array $row, string $patrol_type) {
+        foreach ($row as $key => $value) {
+            $this->$key = $value;
+        }
+        $this->patrol_type = $patrol_type;
+    }
+    public function setLocation(System $system) {
```

Comment:
I think `setLocation` makes more sense here since it only sets these values in the end:
```
$this->current_x = $position['x'];
$this->current_y = $position['y'];
```